imap_data_access/file_validation.py

```diff
@@ -444,11 +542,122 @@ def construct_path(self) -> Path:
             Upload path
         """
         spice_dir = imap_data_access.config["DATA_DIR"] / "spice"
-        subdir = _SPICE_DIR_MAPPING[self.file_extension]
+        subdir = _SPICE_DIR_MAPPING[self.spice_metadata["type"]]
         # Use the file suffix to determine the directory structure
         # IMAP_DATA_DIR/spice/<subdir>/filename
         return spice_dir / subdir / self.filename
 
+    @staticmethod
+    def _matches_on_group(
```

Comment:
I think if you combine all the regexes with "|", then you can just call `regex.match()` on that directly and don't need this function.